pkg/framework/framework.go

```diff
@@ -125,13 +125,21 @@ func LoadClientset() (*kubernetes.Clientset, error) {
 }
 
 func IsStatusAvailable(client runtimeclient.Client, name string) bool {
+	return expectStatusAvailableIn(client, name, WaitShort)
+}
+
+func WaitForStatusAvailable(client runtimeclient.Client, name string) bool {
+	return expectStatusAvailableIn(client, name, WaitMedium)
+}
```

Comment:
Do we need two helpers with different names like this? This to me implies that one is sync and one is async, but they are both async. Can we make this clearer somehow?

Reply:
It is essentially like this. One scenario is expecting an absence of interruption in MAO `IsStatusAvailable` with minimal amount of polling (just to exclude case when is something with API server), and another is actually waiting for Available condition in ~3 min, which is the time required to roll out our components and verify that they are stable.

Reply:
Why not name them `WaitForStatusAvailableShort`/`WaitForStatusAvailableLong` in that case? I still think it's confusing that these are basically the same but have different names completely

Reply:
Updated PTAL @JoelSpeed